versioned/spi/src/main/java/org/projectnessie/versioned/VersionStore.java

```diff
@@ -315,4 +316,7 @@ Stream<Commit<METADATA, VALUE>> getCommits(Ref ref, boolean fetchAdditionalInfo)
    * @return A stream of reflog entries.
    */
   Stream<RefLogDetails> getRefLog(Hash refLogId) throws RefLogNotFoundException;
+
+  /** Get all unreachable heads by reading all the commit logs. */
+  Set<Hash> getUnreachableHeads(long expectedCommitCount) throws ReferenceNotFoundException;
```

Comment:
--> Stream

Reply:
As pagination is not supported, do we still need it as stream wrapping the HashSet?